### PR TITLE
Feature/1 initial version

### LINKED ISSUE #1: Initial version
Create an initial version of this action.

### PR DESCRIPTION
**Pull request by miroslavpojer**
Initial implementation of Action.

Closes #1

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Expected behavior
+A clear and concise description of what you expected to happen.
+
+## Screenshots
+If applicable, add screenshots to help explain your problem.
+
+## Desktop (please complete the following information):
+ - OS: [e.g. iOS]
+ - Browser [e.g. chrome, safari]
+ - Version [e.g. 22]
```
Comment (Zejnilovic) [inline .github/ISSUE_TEMPLATE/bug_report.md:27]: Does this make sense for a GH action?

File: .idea/.gitignore
Comment (Zejnilovic) [inline .idea/.gitignore:1]: Why is this here? There is already a `.gitignore` in root path

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# How to Contribute?
+
+## **Identifying and Reporting Bugs**
+* **Ensure the bug has not already been reported** by searching our **[GitHub Issues](https://github.com/AbsaOSS/generate-release-notes/issues)**.
+* If you cannot find an open issue describing the problem, use the **Bug report** template to open a new one. Tag it with the **bug** label.
+
+## **Proposing New Features**
+
+* **Check if the feature has already been requested** by searching through our **[GitHub Issues](https://github.com/AbsaOSS/enceladus/issues)**.
```
Comment (Zejnilovic) [inline CONTRIBUTING.md:9]:
```suggestion
* **Check if the feature has already been requested** by searching through our **[GitHub Issues](https://github.com/AbsaOSS/generate-release-notes/issues)**.
```

File: README.md
Comment (Zejnilovic) [inline README.md:1]: We will need to add motivation and why this is "better" then https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

File: README.md [hunk 3]
```diff
@@ -1,2 +1,119 @@
...
-Efficiently automate your release note generation with 'generate-release-notes'. This tool scans your target GitHub repository's issues, sorting and organizing them into well-formatted release notes. Perfect for maintaining a streamlined and organized release process.
+# Generate Release Notes Action
+
+This GitHub Action automatically generates release notes for a given release tag by categorizing contributions into user-defined chapters based on labels. It streamlines the process of documenting new features, bug fixes, and breaking changes in your project releases.
+
+## Prerequisites
+
+Before you begin, ensure you have:
+- A GitHub account with appropriate permissions to access the repository.
+- A GitHub token with permissions to fetch repository data.
```
Comment (Zejnilovic) [inline README.md:9]: An account is not needed. The token can be associated with an organization.

```suggestion
Before we begin, ensure you have a GitHub Token with permission to fetch repository data such as Issues and Pull Requests.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,119 @@
...
+### Adding the Action to Your Workflow
+
+Add the following step to your GitHub workflow:
+
+```yaml
+- name: Generate Release Notes
+  uses: AbsaOSS/generate-release-notes@0.1.0
+  with:
+    github_token: ${{ secrets.GITHUB_TOKEN }}
+    release_name: v${{ steps.set_tag.outputs.tag }}
```
Comment (Zejnilovic) [inline README.md:22]: Don't force your implementation and use basics for examples

```suggestion
    release_name: v${{ inputs.release_name }}
```

**Review by Zejnilovic — COMMENTED** [review]
Comments about the structure and docs.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (miroslavpojer) [reply .github/ISSUE_TEMPLATE/bug_report.md:27]: No. You are right. Will remove it.

File: .idea/.gitignore
Comment (miroslavpojer) [reply .idea/.gitignore:1]: This file should not be here. Removed.

File: CONTRIBUTING.md (hunk 2 above)
Comment (miroslavpojer) [reply CONTRIBUTING.md:9]: Thanks. Missed this one.

File: action.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+name: 'Generate Release Notes'
+description: 'Automatically generates release notes for a given release tag, categorized into chapters based on labels.'
+inputs:
+  repo:
```
Comment (Zejnilovic) [inline action.yml:4]: You don't need repo, use `const github = require('@actions/github');` and `github.context`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,117 @@
...
+### Adding the Action to Your Workflow
+
+Add the following step to your GitHub workflow:
+
+```yaml
+- name: Generate Release Notes
+  uses: AbsaOSS/generate-release-notes@0.1.0
+  with:
+    github_token: ${{ secrets.GITHUB_TOKEN }}
+    release_name: v${{ inputs.release_name }}
```
Comment (Zejnilovic) [inline README.md:20]: remove release name

File: README.md [hunk 7]
```diff
@@ -1,2 +1,117 @@
...
+
+## Setup and Usage
+
+### Adding the Action to Your Workflow
+
+Add the following step to your GitHub workflow:
+
+```yaml
+- name: Generate Release Notes
+  uses: AbsaOSS/generate-release-notes@0.1.0
```
Comment (Zejnilovic) [inline README.md:17]:
```suggestion
- name: Generate Release Notes
  id: release_notes
  uses: AbsaOSS/generate-release-notes@0.1.0
```

File: README.md
Comment (miroslavpojer) [reply README.md:1]: Prepared new chapter "Motivation" in README.md.

File: action.yml (hunk 5 above)
Comment (miroslavpojer) [reply action.yml]: Refactored.

File: README.md (hunk 6 above)
Comment (miroslavpojer) [reply README.md]: Removed.

File: README.md (hunk 7 above)
Comment (miroslavpojer) [reply README.md:25]: Changed.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,121 @@
...
+  with:
+    github_token: ${{ secrets.GITHUB_TOKEN }}
+    chapters: '[
+      {"title": "Breaking Changes 💥", "label": "breaking-change"},
+      {"title": "New Features 🎉", "label": "enhancement"},
+      {"title": "Bugfixes 🛠", "label": "bugfix"}
+    ]'
+    warnings: true
+```
```
Comment (Zejnilovic) [inline README.md:31]: Please add echo example of printing out to STDOUT the release notes

**Review by Zejnilovic — APPROVED** [review]

File: README.md (hunk 8 above)
Comment (miroslavpojer) [reply README.md]: Added in commit - 42931a69714b0b9afc483184c985938d9bc67113.

**Review by Zejnilovic — APPROVED** [review]
